src/vm/assembly.cpp

```diff
@@ -1965,6 +1965,10 @@ INT32 Assembly::ExecuteMainMethod(PTRARRAYREF *stringArgs, BOOL waitForOtherThre
     IfFailThrow(hr);
     
     END_ENTRYPOINT_THROWS;
+
+    // Sync the latched exit code with the return value of main function.
+    SetLatchedExitCode(iRetVal);
```

Comment:
> Is this really needed?

The condition in this line will be false and let the latched exit code keep 0 at least for the example [here](https://github.com/dotnet/coreclr/issues/6206#issuecomment-231935917):
https://github.com/mazong1123/coreclr/blob/0c46640307dab1d8359f4e3b856a4390d9a6389d/src/vm/assembly.cpp#L1852

I don't know why we're checking the stringArgs here. Seems like we should remove the check. Do you have any idea about this line of code?